### DISCUSSION TITLE
Solax Modbus Doku 3.34

### DISCUSSION THREAD

**Discussion by HA-Supi** [wills106/homeassistant-solax-modbus#1115] [Show and tell]
See attached …. 
[Solax_Hybrid_X1&X3-G4_ModbusTCP&RTU_V3.34_-_English_240313.pdf](https://github.com/user-attachments/files/17530189/Solax_Hybrid_X1.X3-G4_ModbusTCP.RTU_V3.34_-_English_240313.pdf)

**Comment by PatrikTrestik** [comment]
@wills106 do you have newer version than 3.34?
Can you share, please?

**Reply by wills106** [reply]
I missed this version. It's actually newer than what I had.

Looks like there is a newer EVC doc as well as this v3.334 hints at being able to write to the EV Charger.

**Reply by Larasik** [reply]
[(Solax)Hybrid_X1&X3-G4_ModbusTCP&RTU_V3.36-English_240611[1].pdf](https://github.com/user-attachments/files/19076476/Solax.Hybrid_X1.X3-G4_ModbusTCP.RTU_V3.36-English_240611.1.pdf)

**Comment by PatrikTrestik** [comment]
I have this for EVC G1: [EvCharger.ModbusRTU.V3.5.pdf](https://github.com/user-attachments/files/18746085/EvCharger.ModbusRTU.V3.5.pdf)
And this for EVC G2: [X1 X3-HAC EV Charger Modbus TCP&RTU V1.0.0-EN.pdf](https://github.com/user-attachments/files/18746087/X1.X3-HAC.EV.Charger.Modbus.TCP.RTU.V1.0.0-EN.pdf)

Latest i have. But I think that for Inverter there has to be newer than 3.34. But could not find it.

**Reply by wills106** [reply]
Thanks, looking at the v3.5 doc. Have they now added Modbus TCP over the WiFi connection?

**Reply by PatrikTrestik** [reply]
My EVC does not respond to TCP 502. But I have to check if I have latest Firmware. Will check later.

**Comment by PatrikTrestik** [comment]
I can confirm that reading and writing EVC through Invertor works.
Chapter "Read or Set the EVC register via Inverter" is little misleading. See my test example for clarity:
- Want to read EVC Register 0x060D "DeviceMode"
- **Substract EVC base address 0x0600** - not mentioned in doc.
- Add 0x1000 as per documentation
- Read (FN 0x3) register 0x100D = success
- DeviceMode is not good test register. It is always 0 untill car is connected!
- Write (FN 0x10) works too.
- Reading input registers (FN 0x4) works too. Here base is 0x0 so simply add 0x1000.

Happy modbussing.

**Comment by lieglw** [comment]
I as described in an other conversation. When I am changing the Charger Use Mode (Fast, Green, ECO, Stop) nothing happens. By using "start charging" from control command the Charger starts, nether the Use Mode is change to Fast nor the run mode. It stays at preparing. So could it be that Charge use Mode Drop Down uses the wrong register?
![image](https://github.com/user-attachments/assets/b5e29db1-c129-4f5f-8229-7f0da3a1cc5e)
![image](https://github.com/user-attachments/assets/81f14430-9c11-42aa-b941-b7bcab535a33)

And also for the state:
![image](https://github.com/user-attachments/assets/55c958fa-2822-4d22-80ef-4dbfa7ceb7ec)
![image](https://github.com/user-attachments/assets/5cfa3708-2311-482f-a396-be5d01ce218b)

Could this also somehow implemented for the different EVCs:
![image](https://github.com/user-attachments/assets/434ae48d-2206-45a3-b7ef-c30a881ce7ec)

**Comment by stuart-jjj** [comment]
Hi everyone, can anyone help me source the modbus TCP doc for the X1 G3 hybrid range (I have an X1 G3 retrofit)? 

Solax provided the attached doc when I asked but as you can see by comparing to other docs for the G4 range the write-once and write-many register tables are not present so it is no use for development. They say that's all they can find, which seems hard to believe. 

[Solax External Communication Protocol Hybrid X1&X3-G3 ModbusTCP&RTU V3.28.pdf](https://github.com/user-attachments/files/19558837/Solax.External.Communication.Protocol.Hybrid.X1.X3-G3.ModbusTCP.RTU.V3.28.pdf)
